### ISSUE TITLE
Rust 1.73 + emscripten 3.1.44 fails if using filesystem api

### ISSUE THREAD

**Issue by selassje** [rust-lang/rust#116655]
It builds fine with EMSDK 3.1.43.

```rust
use std::{fs::File, io::Read};

fn main() {
    let mut file = File::open("Hello world!").unwrap();
    let mut buf = Vec::new();
    file.read_to_end(&mut buf);
}
```

  = note: wasm-ld: error: C:\Users\Przemyslaw\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\wasm32-unknown-emscripten\lib\libstd-330edb0856239641.rlib(std-330edb0856239641.std.fa9f9c7c191d43dc-cgu.0.rcgu.o): undefined symbol: open64
          wasm-ld: error: C:\Users\Przemyslaw\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\wasm32-unknown-emscripten\lib\libstd-330edb0856239641.rlib(std-330edb0856239641.std.fa9f9c7c191d43dc-cgu.0.rcgu.o): undefined symbol: fstat64
          wasm-ld: error: C:\Users\Przemyslaw\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\wasm32-unknown-emscripten\lib\libstd-330edb0856239641.rlib(std-330edb0856239641.std.fa9f9c7c191d43dc-cgu.0.rcgu.o): undefined symbol: lseek64

### Meta

`rustc --version --verbose`:
```
rustc 1.73.0 (cc66ad468 2023-10-03)
binary: rustc
commit-hash: cc66ad468955717ab92600c770da8c1601a4ff33
commit-date: 2023-10-03
host: x86_64-pc-windows-msvc
release: 1.73.0
LLVM version: 17.0.2
```

<details><summary>Backtrace</summary>
<p>

```
<backtrace>
```

</p>
</details>

**Comment by kleisauke** [comment]
This should be fixed in Emscripten 3.1.46, see: https://github.com/emscripten-core/emscripten/pull/20123.

Compatibility with Emscripten 3.1.44 and 3.1.45 is addressed in PR https://github.com/rust-lang/libc/pull/3308, which is included in libc 0.2.149. This should be available in Rust 1.75.0 (see #116527).

**Comment by tari** [comment]
I don't think newer emscripten fixed this, or if it was fixed it regressed again. I have emscripten 3.1.48 and the reproducer from the original report above still fails to link with the same undefined symbols.

Building with rustc 1.75 (I tried 1.75.0-beta.5) does work, though.

**Comment by lonsbio** [comment]
Just to add that using `1.75.0-beta.5` as advised by @tari was the only way to fix this issue for me

```
rustc 1.75.0-beta.5 
emcc 3.1.50
```

**Comment by kleisauke** [comment]
Indeed, I can confirm that it's still broken on the latest stable version of Rust in combination with Emscripten 3.1.44 or higher. I've updated my previous comment.
<details>
  <summary>Details</summary>

```console
$ curl https://sh.rustup.rs -sSf | sh -s -- -y --profile minimal --target wasm32-unknown-emscripten
$ source "$HOME/.cargo/env"
$ rustc --version --verbose
rustc 1.74.1 (a28077b28 2023-12-04)
binary: rustc
commit-hash: a28077b28a02b92985b3a3faecf92813155f1ea1
commit-date: 2023-12-04
host: x86_64-unknown-linux-gnu
release: 1.74.1
LLVM version: 17.0.4
$ emcc -dumpversion
3.1.51
$ cat <<EOT > main.rs
use std::{fs::File, io::Read};

fn main() {
    let mut file = File::open("Hello world!").unwrap();
    let mut buf = Vec::new();
    let _ = file.read_to_end(&mut buf);
}
EOT
$ rustc main.rs --target wasm32-unknown-emscripten
error: linking with `emcc` failed: exit status: 1
...
```
</details>

This is curious, as the above mentioned Emscripten PR is supposed to fix that, at least for C/C++ programs.
